File: Practice/07/Python/area_of_a_triangle.py
#!/usr/bin/python3
from math import sqrt

# Задание 7 - Площадь триугольника


class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def length(self, point):
        return sqrt((point.x - self.x)**2 + (point.y - self.y)**2)


def sides_area():
    print("--> Через длины сторон: ")
    a = int(input("Введите сторону a: "))
    b = int(input("Введите сторону b: "))
    c = int(input("Введите сторону c: "))
    p = (a + b + c) / 2
    s = sqrt(p * (p - a) * (p - b) * (p - c))
    return s


def coord_area():
    print("--> Через координаты вершин: ")
    x1, y1 = map(int, input('Координаты первой вершины: ').split(' '))
    x2, y2 = map(int, input('Координаты второй вершины: ').split(' '))
    x3, y3 = map(int, input('Координаты третьей вершины: ').split(' '))
    A = Point(x1, y1)
    B = Point(x2, y2)
    C = Point(x3, y3)
    p = (A.length(B) + A.length(C) + B.length(C)) / 2
    S = sqrt(p*(p - A.length(B))*(p - A.length(C))*(p - B.length(C)))
    return S


if __name__ == "__main__":
    print("7. Площадь триугольника")
    t = int(input("Введите способ решения (1 или 2): "))
    if t == 1:
        print("S: {:.2f}".format(sides_area()))
    elif t == 2:
        print("S: {:.2f}".format(coord_area()))
    else:
        print("[Ошибка] Доступные значение 1 и 2")

File: Practice/07/C++/area_of_a_triangle.cpp
#include <iostream>
#include <cmath>

/* Задание 7 - Площадь триугольника */

using namespace std;

struct point
{
    float x;
    float y;
    float length(point pt)
    {
        return sqrt(pow(pt.x - x, 2.0f) + pow(pt.y - y, 2.0f));
    }
};

float sides_area()
{
    cout << "--> Через длины сторон: " << endl;
    float a, b, c, p, s;
    cout << "Введите сторону a: ";
    cin >> a;
    cout << "Введите сторону b: ";
    cin >> b;
    cout << "Введите сторону c: ";
    cin >> c;
    p = (a + b + c) / 2;
    s = p * (p - a) * (p - b) * (p - c);
    return sqrt(s);
}

float coord_area()
{
    cout << "--> Через координаты вершин: " << endl;
    float x1, y1, x2, y2, x3, y3;
    cout << "Координаты первой вершины: ";
    cin >> x1 >> y1;
    cout << "Координаты второй вершины: ";
    cin >> x2 >> y2;
    cout << "Координаты третьей вершины: ";
    cin >> x3 >> y3;
    point A = {x1, y1};
    point B = {x2, y2};
    point C = {x3, y3};
    float p = (A.length(B) + A.length(C) + B.length(C)) / 2;
    float S = sqrt(p * (p - A.length(B)) * (p - A.length(C)) * (p - B.length(C)));
    return S;
}

int main()
{
    cout << "7. Площадь триугольника" << endl;
    cout << "Введите способ решения (1 или 2): ";
    int type = 1;
    cin >> type;
    if (type == 1)
        cout << "S: " << sides_area() << endl;
    else if (type == 2)
        cout << "S: " << coord_area() << endl;
    else
        cout << "[Ошибка] Доступные значение 1 и 2" << endl;
    return 0;
}